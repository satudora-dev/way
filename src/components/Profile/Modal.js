import React from 'react';
import Button from '@material-ui/core/Button';
import Modal from '@material-ui/core/Modal';
import Grid from '@material-ui/core/Grid';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import aitl_positions from '../../components/aitl_positions';
import aitl_projects from '../../components/aitl_projects';

const OriginalModal = ({
  children,
  mode,
  buttonText,
  modalOpen,
  currentChecks,
  updateData,
  onModalClose,
  profileUserKey
}) => {
  return (
    <div>
      <Grid>
        <Modal
          open = {modalOpen}
          onClose = {() => onModalClose()}
        >
          <div>
            {children}
          </div>

          <Button onClick = {() => onModalClose()}>
            {buttonText}
          </Button>
        </Modal>
      </Grid>
    </div>
  )
}

export default OriginalModal
