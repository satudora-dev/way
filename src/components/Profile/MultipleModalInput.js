import React from 'react';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import Checkbox from '@material-ui/core/Checkbox';
import ListItemText from '@material-ui/core/ListItemText';
import aitl_positions from '../../components/aitl_positions';
import aitl_projects from '../../components/aitl_projects';


const MultipleModalInput = ({
  currentChecks,
  inputArray,
  updateData,
  profileUserKey
}) => {
  return (
    <Select
      multiple
      value = {currentChecks}
      renderValue={value => value.join(', ')}
      onChange = {(e) => updateData(currentChecks, e.target.value, profileUserKey)}
    >
      {inputArray.map(
        choice => (
          <MenuItem
            key = {choice}
            value = {choice}
          >
            <Checkbox
              checked = {currentChecks.indexOf(choice) !== -1}
            />
            <ListItemText
              primary = {choice}
            />
          </MenuItem>
        ))}
    </Select>
  )
}

export default MultipleModalInput
