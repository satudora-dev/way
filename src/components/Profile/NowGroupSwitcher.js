import React, { Component } from 'react';

import Button from '@material-ui/core/Button';
import Checkbox from '@material-ui/core/Checkbox';
import FormGroup from '@material-ui/core/FormGroup';
import nowGroupList from '../nowGroupList';

const style = {
  categoryStyle: {
    textAlign: "left",
    margin: "10px 30px",
    fontFamily: "Avenir",
  },
  tagStyle: {
    textAlign: "left",
    fontFamily: "Avenir",
    margin: "10px 30px",
    marginBottom: "40px",
  },
  btnStyle: {
    marginRight: "10px",
    marginBottom: "10px",
    backgroundColor: "#04B486",
    color: "white",
    textTransform: "none",
  },
  disabledBtnStyle: {
    marginRight: "10px",
    marginBottom: "10px",
    textTransform: "none",
    color: 'black'
  },
}

export default class NowGroupSwitcher extends Component {
  constructor(props) {
    super(props);

    if (!props.nowGroup) {
      let initialNowGroup = {};
      for (let name of nowGroupList) {
        initialNowGroup[name] = false;
      }
      props.initializeNowGroup(initialNowGroup);
    }
  }

  onChange(nowName) {
    let nextNowGroup = this.props.nowGroup;
    nextNowGroup[nowName] = !this.props.nowGroup[nowName];
    this.props.updateNowGroup(nextNowGroup);
  }

  toNowPage(nowName) {
    this.props.history.push(`../users?now=${nowName}`);
  }

  render() {
    let nowGroup = this.props.nowGroup;
    return (
      <div className="NowSwitch">
        <h3 style={style.categoryStyle}>NOW</h3>
        <hr />
        {(() => {
          if (nowGroup) {
            return (
              <div style={style.tagStyle}>
                <FormGroup row>
                  {nowGroupList.map((item, i) => {
                    return (
                      <div key={i}>
                        {(() => {
                          if (this.props.canEdit) {
                            return(
                            <Checkbox
                              checked={nowGroup[item]}
                              onChange={() => this.onChange(item)}
                            />);
                          }
                        })()}
                        <Button
                          variant="contained"
                          disabled={!nowGroup[item]}
                          onClick={() => this.toNowPage(item)}
                          style={(() => {
                            if (nowGroup[item]) {
                              return style.btnStyle;
                            }
                            else {
                              return style.disabledBtnStyle;
                            }
                          })()}>
                          {item}
                        </Button>
                      </div>);
                  })}
                </FormGroup>
              </div>
            );
          }
        })()}
      </div>
    );
  }
}