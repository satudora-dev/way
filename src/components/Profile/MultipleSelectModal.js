import React from 'react';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import Button from '@material-ui/core/Button';
import Modal from '@material-ui/core/Modal';
import Grid from '@material-ui/core/Grid';
import Checkbox from '@material-ui/core/Checkbox';
import ListItemText from '@material-ui/core/ListItemText';
import TextField from '@material-ui/core/TextField';


const style = {
  btnstyle: {
    marginRight: "10px",
    marginBottom: "10px",
    backgroundColor: "#04B486",
    "color": "white",
    textTransform: "none",
  },
  disabledstyle: {
    marginRight: "10px",
    marginBottom: "10px",
    backgroundColor: "gray",
    "color": "white",
    textTransform: "none",
  },
}

const MultipleSelectModal = ({
  children,
  buttonText,
  modalOpen,
  inputArray,
  currentChecks,
  updateData,
  onModalClose,
  profileUserKey,
  ...other
}) => {
  return (
    <div>
      <Grid>
        <Modal
          open = {modalOpen}
          onClose = {() => onModalClose()}
        >

          <div {...other}>
            <div>
              <h3>{children}</h3>
            </div>
            <div>
              <Select
                multiple
                value = {currentChecks}
                renderValue={value => value.join(', ')}
                onChange = {(e) => updateData(currentChecks, e.target.value, profileUserKey)}
              >
                {inputArray.map(
                  choice => (
                    <MenuItem
                      key = {choice}
                      value = {choice}
                    >
                      <Checkbox
                        checked = {currentChecks.indexOf(choice) !== -1}
                      />
                      <ListItemText
                        primary = {choice}
                      />
                    </MenuItem>
                  ))}
              </Select>
            </div>
            <div
              style = {{margin:"10px"}}
            >
              <Button
                style = {style.btnstyle}
                variant = "outlined"
                onClick = {() => {
                    onModalClose()
                  }
                }
              >
                {buttonText}
              </Button>
            </div>
          </div>

        </Modal>
      </Grid>
    </div>
  )
}

export default MultipleSelectModal
