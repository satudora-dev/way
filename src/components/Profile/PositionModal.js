import React from 'react';
import Button from '@material-ui/core/Button';
import Modal from '@material-ui/core/Modal';
import Grid from '@material-ui/core/Grid';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import aitl_positions from '../../components/aitl_positions';

const style = {
  btnstyle: {
    marginRight: "10px",
    marginBottom: "10px",
    backgroundColor: "#04B486",
    "color": "white",
    textTransform: "none",
  },
  selectProjectModalStyle: {
    display:"inline-block",
    marginTop:"250px",
    height:"200px",
    //widthはGridでレスポンシブに
    minWidth: "300px",
    backgroundColor:"white",
    textAlign:"center",
    "outline":"none",
    borderRadius:"30px",
    fontFamily:"Avenir",
  },
  disabledstyle: {
    marginRight: "10px",
    marginBottom: "10px",
    backgroundColor: "gray",
    "color": "white",
    textTransform: "none",
  },
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  formControl: {
    margin: 10,
    minWidth: 120,
    maxWidth: 300,
  },
}
const ITEM_HEIGHT = 48;
const ITEM_PADDING_TOP = 8;
const MenuProps = {
  PaperProps: {
    style: {
      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
      width: 500,
    },
  },
};
const PositionModal = ({positionModalOpen, currentPosition, updatePosition, onPositionModalClose, profileUserKey}) => {

  return(
    <div>
      <Grid>
        <Modal open={positionModalOpen}
               onClose={() => onPositionModalClose()}>
          <div style={style.selectProjectModalStyle}>
            <h3>select project</h3>
            <div style={style.root}>
              <Grid
                container
                direction="row"
                justify="center"
                alignItems="center"
              >
                <FormControl style={style.formControl}>
                  <InputLabel htmlFor="select-multiple-checkbox">Position</InputLabel>
                  <Select
                    value={currentPosition}
                    onChange={(e) => updatePosition(currentPosition,e.target.value, profileUserKey)}
                    input={<Input id="position" />}
                    MenuProps={MenuProps}
                  >
                    {aitl_positions.map(project => (
                      <MenuItem key={project} value={project}>{project}</MenuItem>
                    ))}
                  </Select>
                </FormControl>
              </Grid>
            </div>
            <Button style={currentPosition === "" || currentPosition === undefined ? style.disabledstyle : style.btnstyle}
                variant="outlined"
                value="add"
                disabled={currentPosition === "" || currentPosition === undefined}
                onClick={() => onPositionModalClose()}
            >
            done
            </Button>
          </div>
        </Modal>
      </Grid>
    </div>
  )
}

export default PositionModal;
