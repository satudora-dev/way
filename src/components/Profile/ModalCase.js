import React from 'react';
import Button from '@material-ui/core/Button';
import Modal from '@material-ui/core/Modal';
import Grid from '@material-ui/core/Grid';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import NativeSelect from '@material-ui/core/NativeSelect';
import Checkbox from '@material-ui/core/Checkbox';
import ListItemText from '@material-ui/core/ListItemText';
import TextField from '@material-ui/core/TextField';
import aitl_positions from '../../components/aitl_positions';
import aitl_projects from '../../components/aitl_projects';
import SingleModalInput from './SingleModalInput';
import MultipleModalInput from './MultipleModalInput';
import TextModalInput from './TextModalInput';

const style = {
  btnstyle: {
    marginRight: "10px",
    marginBottom: "10px",
    backgroundColor: "#04B486",
    "color": "white",
    textTransform: "none",
  },
  disabledstyle: {
    marginRight: "10px",
    marginBottom: "10px",
    backgroundColor: "gray",
    "color": "white",
    textTransform: "none",
  },
}

const ModalCase = ({
  children,
  modalInput,
  buttonText,
  modalOpen,
  inputArray,
  currentChecks,
  updateData,
  onModalClose,
  profileUserKey,
  addText,
  ...other
}) => {

  let ModalInput
  let input

  return (
    <div>
      <Grid>
        <Modal
          open = {modalOpen}
          onClose = {() => onModalClose()}
        >

          <div {...other}>
            <div>
              <h3>{children}</h3>
            </div>
            <div>
              {ModalInput}
            </div>
            <div
              style = {{margin:"10px"}}
            >
              <Button
                style = {style.btnstyle}
                variant = "outlined"
                onClick = {() => {
                  if (mode === "text") {
                    if (input === "" || input === undefined) {
                      return
                    } else {
                      addText(input, profileUserKey)
                      onModalClose()
                    }
                  } else {
                    onModalClose()
                  }
                }}
              >
                {buttonText}
              </Button>
            </div>
          </div>

        </Modal>
      </Grid>
    </div>
  )
}

export default ModalCase
