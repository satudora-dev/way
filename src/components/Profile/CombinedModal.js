import React from 'react';
import SingleModalInput from './SingleModalInput';
import MultipleModalInput from './MultipleModalInput';
import TextModalInput from './TextModalInput';
import ModalCase from './ModalCase';


const CombinedModal = ({
  children,
  mode,
  buttonText,
  modalOpen,
  inputArray,
  currentChecks,
  updateData,
  onModalClose,
  profileUserKey,
  addText,
  ...other
}) => {

  let ModalInput
  let input

  if (mode === "single") {
    ModalInput = (
      <SingleModalInput
        currentChecks = {currentChecks}
        inputArray = {inputArray}
        updateData = {updateData}
        profileUserKey = {profileUserKey}
      />
    )

  } else if (mode === "multiple") {
    ModalInput = (
      <MultipleModalInput
        currentChecks = {currentChecks}
        inputArray = {inputArray}
        updateData = {updateData}
        profileUserKey = {profileUserKey}
      />
    )
  } else if (mode === "text") {

    const onTextChange = (e) => {
      return (
        e.target.value
      )
    }
    ModalInput = (
      <TextModalInput
        input = {input}
        onTextChange = {onTextChange}
      />
    )
  }

  return (
    <ModalCase
      mode = {mode}
      input = {input}
      modalInput = {ModalInput}
      buttonText = {buttonText}
      modalOpen = {modalOpen}
      inputArray = {inputArray}
      currentChecks = {currentChecks}
      updateData = {updateData}
      onModalClose = {onModalClose}
      profileUserKey = {profileUserKey}
      addText = {addText}
      {...other}
    >
      {children}
    </ModalCase>
  )
}

export default CombinedModal
