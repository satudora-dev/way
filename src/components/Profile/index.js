import React, { Component } from 'react';
import Button from '@material-ui/core/Button';

import AddIcon from '@material-ui/icons/Add'
import EditIcon from '@material-ui/icons/Edit'
import CloseIcon from '@material-ui/icons/Close'


import EditableLabel from '../../components/EditableLabel'
import ImageUploader from '../../components/ImageUploader'
import CombinedModal from './CombinedModal'
import aitl_positions from '../../components/aitl_positions';
import aitl_projects from '../../components/aitl_projects';



class Profile extends Component {
  constructor(props){
    super(props);
    this.state={
      positionModalOpen: false,
      projectModalOpen: false,
      tagModalOpen: false,
    };
  }



  toPositionPage(posName){
    this.props.history.push(`../users?position=${posName}`);
  }

  toProjectPage(prjName){
    this.props.history.push(`../users?project=${prjName}`);
  }

  toTagPage(tagName){
    this.props.history.push(`../users?tag=${tagName}`);
  }


  render() {
    const style = {
      divstyle: {
        backgroundImage: "url('/grad.jpg')",
        backendPosition: "center center",
        backgroundRepeat: "no-repeat",
        backgroundSize: "cover",
      },
      namestyle: {
        color: "#D8D8D8",
        fontFamily: "Avenir",
        fontSize: "40px",
        margin: "10px"
      },
      categorystyle: {
        textAlign: "left",
        margin: "10px 30px",
        fontFamily: "Avenir",
      },
      tagstyle: {
        textAlign: "left",
        fontFamily: "Avenir",
        margin: "10px 30px",
        marginBottom: "40px",
      },
      btnstyle: {
        marginRight: "10px",
        marginBottom: "10px",
        backgroundColor: "#04B486",
        "color": "white",
        textTransform: "none",
      },
      tagbtnstyle: {
        paddingRight:"8px",
        marginRight: "10px",
        marginBottom: "10px",
        backgroundColor: "#04B486",
        textTransform: "none",
        "color": "white",
      },
      positionstyle: {
        marginRight: "10px",
        marginBottom: "10px",
        backgroundColor: "#04B486",
        textTransform: "none",
        "color": "white",
      },
      modalStyle: {
        display:"inline-block",
        marginTop:"250px",
        height:"200px",
        //widthはGridでレスポンシブに
        minWidth: "300px",
        backgroundColor:"white",
        textAlign:"center",
        "outline":"none",
        borderRadius:"30px",
        fontFamily:"Avenir",
      }
    }
    const profileUserKey = this.props.profileUserKey;
    const given = this.props.given || "";
    const family = this.props.family || "";
    const icon = this.props.icon || "/portrait.png";
    const position = this.props.position || "";
    const projects = this.props.projects || [];
    const tags = this.props.tags || [];
    const canEdit = profileUserKey === this.props.ownKey;


    return (
      <div className="Profile">
        <div className="Home" style={style.divstyle}>
          <ImageUploader
            iconSrc={icon}
            profileUserKey={profileUserKey}
            canEdit={canEdit}
            updateIcon={this.props.updateIcon}/>
            <EditableLabel
              style={style.namestyle}
              value={[given,family]}
              onEditEnd={(names)=>this.props.editName(names,profileUserKey)}
              canEdit={canEdit}
            />
        </div>

        <div className="Position">
          <h3 style={style.categorystyle}>position</h3>
          <hr />
          <div style={style.tagstyle}>
            <Button
              variant="contained"
              color="primary"
              disabled={(!position)}
              style={style.positionstyle}
              onClick={()=>this.toPositionPage(position)}
            >
              {position}
            </Button>
            {(() => {
              if(canEdit)
                return(
                    <Button mini onClick={() => this.setState({positionModalOpen: true})}
                            variant="fab" style={style.btnstyle}>
                      <EditIcon/>
                    </Button>
                )
            })()}
          </div>

          <CombinedModal
            style = {style.modalStyle}
            mode = "single"
            buttonText = "DONE"
            modalOpen = {this.state.positionModalOpen}
            inputArray = {aitl_positions}
            currentChecks = {position}
            updateData = {this.props.updatePosition}
            onModalClose = {() => this.setState({positionModalOpen: false})}
            profileUserKey={profileUserKey}
          >
            Position
          </CombinedModal>

        </div>
        <div className="Project">
          <h3 style={style.categorystyle}>projects</h3>
          <hr />
          <div style={style.tagstyle}>
            {projects.map((project,i)=>{
              return (
                <Button
                  key={i}
                  variant="contained"
                  color="primary"
                  style={style.positionstyle}
                  onClick={()=>this.toProjectPage(project)}
                  >
                {project}
                </Button>
              );
            })}
            {(() => {
              if(canEdit)
                return(
                    <Button mini onClick={() => this.setState({projectModalOpen: true})}
                            variant="fab" style={style.btnstyle}>
                      <EditIcon/>
                    </Button>
                )
            })()}

            <CombinedModal
              style = {style.modalStyle}
              mode = "multiple"
              buttonText = "DONE"
              modalOpen = {this.state.projectModalOpen}
              inputArray = {aitl_projects}
              currentChecks = {projects}
              updateData = {this.props.updateProjects}
              onModalClose = {() => this.setState({projectModalOpen: false})}
              profileUserKey={profileUserKey}
            >
              Projects
            </CombinedModal>

          </div>
        </div>

        <div className="Others">
          <h3 style={style.categorystyle}>tags</h3>
          <hr />
          <div style={style.tagstyle}>
            {tags.map((tag,i)=>{
             return (
              <Button key={i} variant="contained" style={style.tagbtnstyle}>
                <span onClick={()=>this.toTagPage(tag)}>{[tag]}&nbsp;&nbsp;</span>
                <CloseIcon style={{fontSize : "90%", }} onClick={()=>this.props.deleteTag(tag, profileUserKey)}/>
              </Button>
             );
           })}
            <Button mini onClick={() => this.setState({tagModalOpen: true})}
                    variant="fab" style={style.btnstyle}>
            <AddIcon />
            </Button>

            <CombinedModal
              style = {style.modalStyle}
              mode = "text"
              buttonText = "DONE"
              modalOpen = {this.state.tagModalOpen}
              onModalClose = {() => this.setState({tagModalOpen: false})}
              addText = {this.props.addTag}
              profileUserKey = {profileUserKey}
            >
              Add Tags
            </CombinedModal>

            </div>
        </div>
      </div>
    );
  }
}



export default Profile;
