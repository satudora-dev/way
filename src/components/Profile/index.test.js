import React from 'react';
import Profile from '../Profile';
import renderer from 'react-test-renderer';

test('Profile snapshot', () => {
  const component = renderer.create(
    <Profile />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
