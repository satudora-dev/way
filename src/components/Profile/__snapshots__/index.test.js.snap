// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile snapshot 1`] = `
<div
  className="Profile"
>
  <div
    className="Home"
    style={
      Object {
        "backendPosition": "center center",
        "backgroundImage": "url('/grad.jpg')",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
      }
    }
  >
    <div>
      <div
        style={
          Object {
            "margin": "auto",
            "position": "relative",
            "width": "256px",
          }
        }
      >
        <img
          alt="Loading..."
          src="/portrait.png"
          style={
            Object {
              "borderRadius": "50%",
              "height": "256px",
              "marginTop": "30px",
              "objectFit": "cover",
              "width": "256px",
            }
          }
        />
        <input
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
        <button
          className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-fab-18 MuiButton-mini-23 MuiButton-contained-12 MuiButton-raised-15"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": "#04B486",
              "color": "white",
              "margin": 0,
              "position": "absolute",
              "right": "32px",
              "top": "240px",
            }
          }
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-2"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-30"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-138"
          />
        </button>
      </div>
      <div
        className="Login"
      />
    </div>
    <span
      style={
        Object {
          "color": "#D8D8D8",
          "fontFamily": "Avenir",
          "fontSize": "40px",
          "margin": "10px",
        }
      }
    >
      <span
        onClick={[Function]}
      >
        
         
      </span>
      <span
        onClick={[Function]}
      >
        
         
      </span>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-30"
        focusable="false"
        onClick={[Function]}
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
  </div>
  <div
    className="Position"
  >
    <h3
      style={
        Object {
          "fontFamily": "Avenir",
          "margin": "10px 30px",
          "textAlign": "left",
        }
      }
    >
      position
    </h3>
    <hr />
    <div
      style={
        Object {
          "fontFamily": "Avenir",
          "margin": "10px 30px",
          "marginBottom": "40px",
          "textAlign": "left",
        }
      }
    >
      <button
        className="MuiButtonBase-root-27 MuiButtonBase-disabled-28 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16 MuiButton-disabled-21"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "#04B486",
            "color": "white",
            "marginBottom": "10px",
            "marginRight": "10px",
            "textTransform": "none",
          }
        }
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-2"
        >
          
        </span>
      </button>
      <button
        className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-fab-18 MuiButton-mini-23 MuiButton-contained-12 MuiButton-raised-15"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "#04B486",
            "color": "white",
            "marginBottom": "10px",
            "marginRight": "10px",
            "textTransform": "none",
          }
        }
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-2"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-30"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-138"
        />
      </button>
    </div>
    <div>
      <div
        className=""
      />
    </div>
  </div>
  <div
    className="Project"
  >
    <h3
      style={
        Object {
          "fontFamily": "Avenir",
          "margin": "10px 30px",
          "textAlign": "left",
        }
      }
    >
      projects
    </h3>
    <hr />
    <div
      style={
        Object {
          "fontFamily": "Avenir",
          "margin": "10px 30px",
          "marginBottom": "40px",
          "textAlign": "left",
        }
      }
    >
      <button
        className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-fab-18 MuiButton-mini-23 MuiButton-contained-12 MuiButton-raised-15"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "#04B486",
            "color": "white",
            "marginBottom": "10px",
            "marginRight": "10px",
            "textTransform": "none",
          }
        }
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-2"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-30"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-138"
        />
      </button>
      <div>
        <div
          className=""
        />
      </div>
    </div>
  </div>
  <div
    className="Others"
  >
    <h3
      style={
        Object {
          "fontFamily": "Avenir",
          "margin": "10px 30px",
          "textAlign": "left",
        }
      }
    >
      tags
    </h3>
    <hr />
    <div
      style={
        Object {
          "fontFamily": "Avenir",
          "margin": "10px 30px",
          "marginBottom": "40px",
          "textAlign": "left",
        }
      }
    >
      <button
        className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-fab-18 MuiButton-mini-23 MuiButton-contained-12 MuiButton-raised-15"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "#04B486",
            "color": "white",
            "marginBottom": "10px",
            "marginRight": "10px",
            "textTransform": "none",
          }
        }
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-2"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-30"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-138"
        />
      </button>
      <div>
        <div
          className=""
        />
      </div>
    </div>
  </div>
</div>
`;
